src/blas/backends/cublas/cublas_level1.cpp

```diff
@@ -475,8 +475,8 @@ inline void iamax(const char *func_name, Func func, sycl::queue &queue, int64_t
     });
     // This requires to bring the data to host, copy it, and return it back to
     // the device
-    result.template get_access<sycl::access::mode::write>()[0] = std::max(
-        (int64_t)int_res_buff.template get_access<sycl::access::mode::read>()[0] - 1, int64_t{ 0 });
+    result.template get_host_access()[0] =
```

Comment:
Do you know how `get_host_access()` manages access type, like write/read?

Reply:
I checked the SYCL 2020 specification again. This time I found it (see [here](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:accessor.host.buffer.tags)). Fix this in the commit [0f8ee1d](https://github.com/oneapi-src/oneMKL/pull/287/commits/0f8ee1d01a4e8fcd096195730812f10696b9ccf6).

I misinterpreted [the bottom of page 32](https://indico.cern.ch/event/964635/contributions/4059369/attachments/2121584/3570990/DPC_Introduction_and_Update.pdf) as all access modes are simplified to `get_host_access()`...

Logs for the commit [0f8ee1d](https://github.com/oneapi-src/oneMKL/pull/287/commits/0f8ee1d01a4e8fcd096195730812f10696b9ccf6):
- hipSYCL 0.9.4: [cuBLAS_hipSYCL_test_log.txt](https://github.com/oneapi-src/oneMKL/files/10962311/cuBLAS_hipSYCL_test_log.txt) (100% tests passed, 0 tests failed out of 1893, 1260 tests skipped)
- LLVM: [cuBLAS_llvm_test_log.txt](https://github.com/oneapi-src/oneMKL/files/10962255/cuBLAS_llvm_test_log.txt) (100% tests passed, 0 tests failed out of 1893)

Reply:
Yes, the default is read_write if no tag is specified. The tests were "unfortunately" passing.